_posts/2020-03-28-everyone-is-dumb.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: "Everyone's A Little Bit Stupid, Sometimes"
+date: 2020-03-28 08:00:00-0600
+categories: software programming
+excerpt: We've all spent 3 hours debugging to discover we put a + where a - should be. In this post I share some of my facepalm moments.
+---
+
+It's really easy to be hard on yourself when you make a mistake, even easier when it's a ridiculously stupid mistake. You might think to yourself:
+
+> Omg, I've been doing this for X years and I just spent hours trying to chase down this stupid bug caused by a misplaced paren! I'm a failure!
+
+But here's a not-so-secret secret. We all do this. I've been doing this for 8 years and I just did it last week. [Scott Hanselman](https://www.hanselman.com/blog/ImAPhonyAreYou.aspx) has been in the industry for over 20 years and *he's* done it. You are not a failure!
+
+To prove it I'll tell you about two of the times I've done this, both in the last month or so.
+
+## None is None
+
+One of the projects I've involved myself in is [Tuxemon](http://www.tuxemon.org), an open source monster-fighting rpg (a Pokemon clone) written in Python. My biggest contribution so far was a major rework of the Item subsystem. Initially the Item model had a list of strings to represent effects, each being simply the name of a method in the Item class that would be called via reflection (python's getattr function). There was no way to pass data to these functions and there was no way to restrict when the player could use the item other than by limiting which game states it could be used from (basically whether or not it could be used in combat).
```

Comment:
```suggestion
One of the projects I've involved myself in is [Tuxemon](http://www.tuxemon.org), an open source monster-fighting rpg (a Pokemon clone) written in Python. My biggest contribution so far has been a major rework of the Item subsystem. Initially the Item model had a list of strings to represent effects, each being simply the name of a method in the Item class that would be called via reflection (python's getattr function). There was no way to pass data to these functions and there was no way to restrict when the player could use the item other than by limiting which game states it could be used from (basically whether or not it could be used in combat).
```